db/system_keyspace.hh

```diff
@@ -354,7 +354,6 @@ public:
 
     typedef std::vector<db::replay_position> replay_positions;
 
-    future<> save_truncation_record(table_id, db_clock::time_point truncated_at, db::replay_position);
```

Comment:
One may argue that this overload is better since it reduces replica::table and system_keyspace interdependencies, but it if we ever decide to carry out this disentanglement we can fix that at that time.